### ISSUE TITLE
Supported HHVM version

### ISSUE THREAD

**Issue by phy25** [splitbrain/dokuwiki#2272]
As [HHVM announces](https://hhvm.com/blog/2017/09/18/the-future-of-hhvm.html) HHVM will not support every PHP7 language feature. And HHVM 3.24 will be the last HHVM release that commits to PHP5 support.

I see a `hhvm` test case (allow failing) in our travis CI. We can either (a) stick that version to 3.24 to prevent future PHP7-style incompatibility in HHVM, or (b) drop HHVM support.

---

By the way, after some research, [our HHVM tests are failing](https://travis-ci.org/splitbrain/dokuwiki/jobs/349989973#L607) because:

- eu.httpbin.org [returns Internal error (80)](https://travis-ci.org/splitbrain/dokuwiki/jobs/349720786#L599) when [SNI domain is not present on client handshake](https://www.ssllabs.com/ssltest/analyze.html?d=eu.httpbin.org&s=54.217.216.34&latest);
- and HHVM [hasn't implemented `peer_name` support in SSL socket](https://github.com/facebook/hhvm/blob/9b8fbb4f63227e3623610ee13574fe8858258107/hphp/runtime/base/ssl-socket.cpp) in 3.24.3 (the latest release), which is unable to send SNI domain data.

Maybe I should file a issue to upstream.

**Comment by micgro42** [comment]
Does hvvm see any actual usage in the wild? [Google trends](https://trends.google.com/trends/explore?q=hvvm) are barely existent and it has only [500 questions on StackOverflow](https://stackoverflow.com/questions/tagged/hhvm). GitHub as a single repo with the [hvvm tag](https://github.com/topics/hvvm).

I would argue for dropping support for hvvm and maybe keeping the tests as allowed failures for "We happen to be compatible with HVVM version x and y, but not after z"

**Comment by splitbrain** [comment]
Yeah, with the release of PHP 7, HHVM no longer offers any significant speed improvements and seems not to gain any serious interest outside of Facebook anymore. I'm happy to drop HHVM testing alltogether (we never officially supported it anyway).
